apps/shipments/iot_client.py

```diff
@@ -36,3 +42,93 @@ def update_shadow(self, device_id, shadow):
             raise AWSIoTError("Error in response from AWS IoT")
 
         return iot_shadow['data']
+
+    def get_list_owner_devices(self, owner_id, next_token=None, results=None, **kwargs):
+        LOG.debug(f'Getting devices for: {owner_id} from AWS IoT')
+        log_metric('transmission.info', tags={'method': 'DeviceAWSIoTClient.get_list_owner_devices'})
+
+        active = kwargs.get('active', None)
+        in_bbox = kwargs.get('in_bbox', None)
+
+        if in_bbox:
+            in_bbox = ','.join([c.strip() for c in in_bbox.split(',')])
+            in_bbox_is_valid, message = self.validate_in_bbox(in_bbox)
+            if not in_bbox_is_valid:
+                raise ParseError(message)
+
+        if not next_token:
+            results = []
+
+        params_dict = OrderedDict(ownerId=owner_id,
+                                  maxResults=settings.IOT_DEVICES_MAX_RESULTS,
+                                  in_bbox=in_bbox if in_bbox else '',
+                                  nextToken=next_token if next_token else '',)
+
+        list_devices = self._get(f'devices?{urlencode(params_dict)}')
+
+        if 'error' in list_devices:
+            raise AWSIoTError("Error in response from AWS IoT")
+
+        new_devices = list_devices['data'].get('devices', None)
+        if new_devices:
+            results.extend(new_devices)
+
+        next_token = list_devices['data'].get('nextToken', None)
+        if next_token:
+            return self.get_list_owner_devices(owner_id, next_token=next_token, results=results, active=active,
+                                               in_box=in_bbox)
+        if active:
+            device_status = active.lower()
+            if device_status == 'true':
+                results = self.filter_list_devices(results)[0]
```

Comment:
- I think `active` should probably be a boolean instead of a string
- `active` defaults to `None` which isn't handled in this elif
- I agree with james, I don't like how filter_list_devices returns a tuple. I think actually filter_list_devices can be reduced to a list comprehension. You don't necessarily have to use this code, but something like that could work I think.
```
        active = False
        filtered_devices = [device for device in list_device 
                            if isinstance(device['shadowData']['reported'], dict) and 
                            (active and is_uuid(device['shadowData']['reported'].get('shipmentId', None)) or
                             not active and not is_uuid(device['shadowData']['reported'].get('shipmentId', None)))
                            ]
```

- But I also think that we should ask Rudny to do this filtering on his end. It's fairly inefficient for us to do it on our end.

Reply:
I talked to Rudney he is going to implement this filter in his lambda.